File: SpeedTestGameClient/core/src/main/java/si/um/feri/speedii/classes/gsonserializers/LocalDateTimeSerializer.java
package si.um.feri.speedii.classes.gsonserializers;

import com.google.gson.*;
import java.lang.reflect.Type;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class LocalDateTimeSerializer implements JsonSerializer<LocalDateTime> {
    @Override
    public JsonElement serialize(LocalDateTime src, Type typeOfSrc, JsonSerializationContext context) {
        if (src != null) {
            return new JsonPrimitive(src.format(DateTimeFormatter.ISO_DATE_TIME));
        }
        return JsonNull.INSTANCE; // handle null values gracefully
    }
}

File: MobileApp/app/src/main/java/sosteric/pora/speedii/localDateTimeGson/LocalDateTimeSerializer.kt
package sosteric.pora.speedii.localDateTimeGson

import com.google.gson.*
import java.lang.reflect.Type
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

class LocalDateTimeSerializer : JsonSerializer<LocalDateTime> {
    override fun serialize(src: LocalDateTime?, typeOfSrc: Type?, context: JsonSerializationContext?): JsonElement {
        return JsonPrimitive(src?.format(DateTimeFormatter.ISO_DATE_TIME))
    }
}
